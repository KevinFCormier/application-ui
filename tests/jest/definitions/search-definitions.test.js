/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'
import {
  getPolicyStatusIcon
} from '../../../src-web/definitions/search-definitions'

describe('search-definitions - getPolicyStatusIcon', () => {
  it('should return compliant status', () => {
    const item = {
      compliant: 'Compliant'
    }
    expect(getPolicyStatusIcon(item)).toMatchSnapshot()
  })
  it('should return NON-compliant status', () => {
    const item = {
      compliant: 'Non-compliant'
    }
    expect(getPolicyStatusIcon(item)).toMatchSnapshot()
  })
  it('should return blank status', () => {
    const item = { }
    expect(getPolicyStatusIcon(item)).toMatchSnapshot()
  })
})
