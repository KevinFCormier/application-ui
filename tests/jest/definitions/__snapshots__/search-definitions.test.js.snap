// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-definitions - getPolicyStatusIcon should return NON-compliant status 1`] = `
<StatusField
  status="critical"
  text="not compliant"
/>
`;

exports[`search-definitions - getPolicyStatusIcon should return blank status 1`] = `"-"`;

exports[`search-definitions - getPolicyStatusIcon should return compliant status 1`] = `
<StatusField
  status="ok"
  text="compliant"
/>
`;
